### ISSUE TITLE
azurerm_kubernetes_cluster - AKS cluster creation fails when load_balancer_profile block not empty and outboundType=userAssignedNATGateway

### ISSUE THREAD

**Issue by humpalu** [hashicorp/terraform-provider-azurerm#16712]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

2.99

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  depends_on              = [azurerm_subnet_nat_gateway_association.aks_nat_gw]
  name                    = var.prefix
  location                = azurerm_resource_group.rg.location
  resource_group_name     = azurerm_resource_group.rg.name
  node_resource_group     = "${var.prefix}-nodes-rg"
  dns_prefix              = var.prefix
  sku_tier                = "Paid"
  private_cluster_enabled = false

  kubernetes_version = data.azurerm_kubernetes_service_versions.current.latest_version

  automatic_channel_upgrade = "patch" 

  auto_scaler_profile {
    balance_similar_node_groups      = true
    max_graceful_termination_sec     = "600"
    scale_down_utilization_threshold = "0.7"
    skip_nodes_with_local_storage    = false # Nodes should use local storage only as cache or temporary not as persistent storage
    skip_nodes_with_system_pods      = false # Pods in kube-system won't block the deployment and must be HA
  }

  addon_profile {
    http_application_routing {
      enabled = false
    }
    kube_dashboard {
      enabled = false
    }
    azure_policy {
      enabled = true
    }
  }

  default_node_pool {
    name                         = "system"
    os_disk_size_gb              = var.system_pool_config.disk_size
    os_disk_type                 = var.system_pool_config.os_disk_type
    vm_size                      = var.system_pool_config.vm_size
    min_count                    = var.system_pool_config.min_count
    max_count                    = var.system_pool_config.max_count
    vnet_subnet_id               = azurerm_subnet.subnet.id
    tags                         = var.tags
    only_critical_addons_enabled = true
    enable_auto_scaling          = true
    enable_host_encryption       = true
    max_pods                     = 30 
    availability_zones           = [1, 2, 3]
    orchestrator_version         = var.system_pool_config.orchestrator_version

    upgrade_settings {
      max_surge = var.system_pool_config.max_surge
    }

  }
  identity {
    type = "SystemAssigned"
  }

  linux_profile {
    admin_username = "kubernetes"
    ssh_key {
      key_data = tls_private_key.node-ssh-key.public_key_openssh
    }
  }

  role_based_access_control {
    enabled = true

    azure_active_directory {
      managed                = "true"
      admin_group_object_ids = var.admin_group_object_ids
    }
  }


  network_profile {
    network_plugin     = "azure"
    network_mode       = "transparent"
    network_policy     = "calico"
    service_cidr       = "10.0.0.0/16"
    dns_service_ip     = "10.0.0.10"
    docker_bridge_cidr = "172.17.0.1/16"
    load_balancer_sku  = "standard"

    load_balancer_profile {
      outbound_ports_allocated  = 0
      idle_timeout_in_minutes   = 4
      managed_outbound_ip_count = 1
    }
    outbound_type = "userAssignedNATGateway"
  }

  tags = var.tags
}
```

### Debug Output/Panic Output

```shell
waiting for creation of Cluster: (Managed Cluster Name "managed-cluster" / Resource Group "managed-cluster-rg"): Code="CreateVMSSAgentPoolFailed" Message="AKS encountered an internal error while attempting the requested Creating operation. AKS will continuously retry the requested operation until successful or a retry timeout is hit. Check back to see if the operation requires resubmission.
```

### Expected Behaviour

The managed cluster is deployed

### Actual Behaviour

Its always failing with the error message above. 
Also ported the to AzureRM 3.5 but the error is the same.

Strange thing, before 1st of May it worked without any problems. After 1st of May it started failing with the error of
`Failure sending request: StatusCode=400 -- Original Error: Code="SubnetsAssociatedWithNATgatewayWhenOutboundTypeIsStandardLoadBalancer"`
thatswhy the `outbound_type = "userAssignedNATGateway"` introduced

### Steps to Reproduce

terraform apply

**Comment by grzeg1** [comment]
Now it's impossible to create a cluster with outbound NAT gateway. Even if forced to create and imported, terraform wants do destroy the cluster and re-create it (then it fails during creation). Seems like a provider bug.

**Comment by grzeg1** [comment]
```
          + load_balancer_profile { # forces replacement
              + effective_outbound_ips    = (known after apply)
              + idle_timeout_in_minutes   = 30
              + managed_outbound_ip_count = (known after apply)
              + outbound_ip_address_ids   = [
                  + "/subscriptions/.../loadbalancer_public_ip",
                ]
              + outbound_ip_prefix_ids    = (known after apply)
              + outbound_ports_allocated  = 0
            }
```

**Comment by grzeg1** [comment]
resource JSON snippet of the semi-created cluster:
```
        "enableRBAC": true,
        "networkProfile": {
            "networkPlugin": "azure",
            "networkPolicy": "calico",
            "loadBalancerSku": "Standard",
            "serviceCidr": "10.0.0.0/16",
            "dnsServiceIP": "10.0.0.10",
            "dockerBridgeCidr": "172.17.0.1/16",
            "outboundType": "userAssignedNATGateway",
            "serviceCidrs": [
                "10.0.0.0/16"
            ],
            "ipFamilies": [
                "IPv4"
            ]
        },
```

**Comment by tothger** [comment]
Hello , 
Thank you for your answer.
I've tried to create the cluster with UserAssigned identity, but it fails at the validation phase

```
identity {
    type = "UserAssigned"
    identity_ids = [azurerm_user_assigned_identity.aks_test.id]
  }
```

Failing here >>

```
$ terraform validate
╷
│ Error: Unsupported argument
│
│   on main.tf line 60, in resource "azurerm_kubernetes_cluster" "aks":
│   60:     identity_ids = [azurerm_user_assigned_identity.aks_test.id]
│
│ An argument named "identity_ids" is not expected here.
```
according to that stackowerflow suggestion: 
https://stackoverflow.com/questions/72184309/azurerm-kubernetes-cluster-aks-encountered-an-internal-error-while-attempting/

**Comment by grzeg1** [comment]
You have a bug in your syntax. Verify with documentation for correct version of your provider.

**Comment by grzeg1** [comment]
We tried this suggestion too but it did not help. I.e cluster creation failed the same way as with system assigned identity.

**Comment by galibozek** [comment]
same here :/

**Comment by grzeg1** [comment]
@tothger can you please rename this issue to something more meaningful like "AKS cluster creation fails when load_balancer_profile block not empty and outboundType=userAssignedNATGateway"

**Comment by grzeg1** [comment]
Thanks. 
We've read the docs again and again and finally it seems the issue may not be what we think.
Why: everything related to the `load_balancer_profile` block is described in the docs as "outbound parameters". Obviously, when using `userAssignedNATGateway` as outbound strategy, loadBalancer strategy does not apply. Thus, `load_balancer_profile` block does not seem to be valid in this context.
If I'm right: the expected bugfix would be to report using `load_balancer_profile` as illegal (syntax level error) when `userAssignedNATGateway` strategy is chosen. Nice to have: change the block name to `load_balancer_outbound_profile`.

**Comment by stephybun** [comment]
Hi @tothger and @grzeg1,

Apologies for the delayed attention this issue received and thank you both for looking into the behaviour here and for leaving your findings in the comments. 

While attempting to reproduce this issue, I didn't receive any errors but Terraform did flag a diff when I tried to set `managed_outbound_ip_count` or `outbound_ip_prefix_ids` within the `load_balancer_profile` block. So the API accepted it as a valid configuration, but didn't actually end up setting these values for the cluster and also didn't return them in the response.

I was however able to set `outbound_ip_address_ids`, `idle_timeout_in_minutes` and `outbound_ports_allocated` within the `load_balancer_profile` successfully, but reading through the AKS docs, this configuration doesn't appear to be applicable or relevant if the cluster is relying on a managed or user defined NAT.

I've raised this issue https://github.com/Azure/azure-rest-api-specs/issues/22469 regarding our findings in the hopes that the AKS team can clarify and/or potentially fix the behaviour of the API. In the meantime I've also opened a PR that will return an error for the invalid configuration as suggested.
